projects/plugins/jetpack/extensions/blocks/opentable/use-restaurant-search.js

```diff
@@ -5,18 +5,23 @@ export const possibleEmbed = /^\s*(http[s]?:\/\/|\<script)/;
 
 export default function useRestaurantSearch( searchTerm, maxResults ) {
 	const [ restaurants, setRestaurants ] = useState( [] );
+	const [ hasRequestFailed, setHasRequestFailed ] = useState( false );
```

Comment:
A basic implementation, but that's sufficient for our case. We might take this further when we know [what solution OpenTable will suggest](https://github.com/Automattic/jetpack/issues/39410#issuecomment-2360232860).